php/composer/Portfile

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        composer composer 2.5.8
+revision            0
+categories          php devel
+platforms           any
+license             MIT
+maintainers         {ego.team:yet @yugaego} openmaintainer
+
+description         a dependency manager for the PHP programming language
+
+long_description    Composer is {*}${description}. It helps you declare, \
+                    manage, and install dependencies of PHP projects.
+
+homepage            https://getcomposer.org
+
+github.tarball_from releases
+distname            ${name}
+extract.suffix      .phar
+
+revision            0
+checksums           rmd160  cbcc680e5206a5f016151ab15824685523538819 \
+                    sha256  f07934fad44f9048c0dc875a506cca31cc2794d6aebfc1867f3b1fbf48dce2c5 \
+                    size    2837394
+
+depends_run         bin:php:php82
+
+extract {
+    file copy ${distpath}/${distfiles} ${workpath}/${distname}
+}
```

Comment:
In its current form, this port copies the distfile to the workpath and then copies it from the workpath to the destroot. You could save time by copying the distfile directly into the destroot in the destroot phase and telling MacPorts to skip the extract phase by writing `extract.only` (with no value).

On the other hand, if the contents of the phar needs to be patched for example to remove auto-updating code or to make it use some specific version of php, then you might want to change the extract phase to unpack the phar so that it can then be patched in the patch phase and modify the destroot phase to create a new phar in the destroot.